src/main/java/dev/railroadide/railroad/ide/ui/MarkdownPreviewPane.java

```diff
@@ -0,0 +1,204 @@
+package dev.railroadide.railroad.ide.ui;
+
+import dev.railroadide.core.ui.RRButton;
+import dev.railroadide.core.ui.RRVBox;
+import dev.railroadide.railroad.Railroad;
+import io.github.raghultech.markdown.javafx.preview.MarkdownWebView;
+import javafx.application.Platform;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.control.ContextMenu;
+import javafx.scene.control.CustomMenuItem;
+import javafx.scene.control.SplitPane;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Priority;
+import javafx.scene.layout.VBox;
+import javafx.scene.web.WebView;
+import org.kordamp.ikonli.Ikon;
+import org.kordamp.ikonli.fontawesome6.FontAwesomeBrands;
+import org.kordamp.ikonli.fontawesome6.FontAwesomeSolid;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class MarkdownPreviewPane extends RRVBox {
+    private final MarkdownWebView preview;
+    private final Path markdownFile;
+
+    private TextEditorPane textEditorPane;
+    private WebView webViewPane;
+
+    private final HBox topRow;
+    private final HBox markdownButtons;
+    private final HBox switchButtons;
+
+    public MarkdownPreviewPane(Path markdownFile) {
+        this.markdownFile = markdownFile;
+
+        try {
+            preview = new MarkdownWebView(Files.readString(markdownFile), Railroad.getHostServicess());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        markdownButtons = createMarkdownButtons();
+        HBox.setHgrow(markdownButtons, Priority.ALWAYS);
+        switchButtons = createViewButtons();
+
+        topRow = new HBox(markdownButtons, switchButtons);
```

Comment:
`RRHBox` should be used over `HBox` where possible.
```suggestion
        topRow = new RRHBox(markdownButtons, switchButtons);
```